src/components/Modal/Votes/VotingModal/VotingModalSurvey.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+
+import Paragraph from 'decentraland-gatsby/dist/components/Text/Paragraph'
+import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+import TokenList from 'decentraland-gatsby/dist/utils/dom/TokenList'
+import { Button } from 'decentraland-ui/dist/components/Button/Button'
+import { Header } from 'decentraland-ui/dist/components/Header/Header'
+import { Modal } from 'decentraland-ui/dist/components/Modal/Modal'
+
+import { Survey, SurveyTopicAttributes } from '../../../../entities/SurveyTopic/types'
+import { formatChoice } from '../../../../modules/votes/utils'
+import { ProposalPageContext, SelectedChoice } from '../../../../pages/proposal'
+import SentimentSurvey from '../../../Proposal/SentimentSurvey/SentimentSurvey'
+import '../../ProposalModal.css'
+
+import './VotingModal.css'
+
+interface VotingModalSurveyProps {
+  survey: Survey
+  setSurvey: React.Dispatch<React.SetStateAction<Survey>>
+  surveyTopics: Pick<SurveyTopicAttributes, 'label' | 'topic_id'>[] | null
+  isLoadingSurveyTopics: boolean
+  onCastVote: (selectedChoice: SelectedChoice, survey?: Survey) => void
+  castingVote: boolean
+  proposalContext: ProposalPageContext
+}
+
+export function VotingModalSurvey({
+  survey,
+  setSurvey,
+  surveyTopics,
+  isLoadingSurveyTopics,
+  onCastVote,
+  castingVote,
+  proposalContext,
+}: VotingModalSurveyProps) {
+  const t = useFormatMessage()
+  const { selectedChoice, showVotingError, retryTimer } = proposalContext
+
+  if (!selectedChoice.choiceIndex || !selectedChoice.choice) {
```

Comment:
I think you can move this check to `VotingModal`

`const showVotingModalSurvey = selectedChoice.choiceIndex && selectedChoice.choice`

Reply:
Ty! Yeah, the check is there but as I am passing the properties to the ModalSurvey inside the proposal context object, the validation isn't taken into account. 

I think we can do `formatChoice(selectedChoice.choice!)` for this one